picolibc.ld.in

```diff
@@ -113,6 +96,28 @@ SECTIONS
 
 	} >flash AT>flash :text
 
+	.rodata : {
+
+		/* read-only data */
+		*(.rdata)
+		*(.rodata .rodata.*)
+		*(.gnu.linkonce.r.*)
+
+		*(.srodata.cst16)
+		*(.srodata.cst8)
+		*(.srodata.cst4)
+		*(.srodata.cst2)
+		*(.srodata .srodata.*)
+
+	} >flash AT>flash :text
+
+	.data.rel.ro : {
+
+		/* relocation information (not generally used with picolibc) */
```

Comment:
Maybe rephrase this as "data that needs relocating" since the relocation information is in .rel(a).dun?

Reply:
sure; I'm pretty unclear on what this section should contain, after all :-)

Reply:
Are you otherwise happy with the change?

Reply:
Yes, everything else looks good to me.

Reply:
In general this section should be empty for most picolibc targets but it will be non-empty for PIC outputs or targets that require relocations for initializing some globals (e.g. CHERI enabled RISCV)